Competitive_Programming/C++/Array/MaxMinProductSubset.md

```diff
@@ -0,0 +1,136 @@
+    temp_container.push_back(subsets);
+    for (int i = position; i < arr.size(); i++)
+    {
+
+        //while iterating check whether to include/not include that element in the final set to be uploaded
+        subsets.push_back(arr[i]);
+
+        // recursive call to get the next element.
+        getsubsets(arr, temp_container, subsets, i + 1);
+
+        //If say we have to exclude some element it triggers a back-track by removing that element
+        subsets.pop_back();
+    }
+    return;
+}
+
+```
+
+As visible each element is check twice so basically 2 recursive calls for its inclusion/exculsion , adding to that the size of the given array hence by basic combinotronics it becomes 2<sub>n</sub> and as another vector is initialized in midway to work on, this increases both the time & space complexity of the whole program overall as :
+Time Complexity : **nO(2<sup>n</sup>)**
+Space Complexity : O(n)
+
+Above explanation puts a clear picture how it would slow down the operation activity of the simple program.
+
+## A Better Approach 💡
+
+A better approach to the given statement would be to using fundamental principle of counting and simple math-theory on the elements and checking teh count of positive or negetive or zeros present in the array and hence proceeding with those use-cases and border-cases accordingly :
+
+#### For maximum product :
+
+**Case 1:** If there are no negetive elements in the array then it is simply the product of all elements such that **array[index]>0**.
+Example:
+arr[] = {2,6,4,5} = (2x6x4x5) = 240
+**Case 2:** If there are some negetive elements in the array then we have some subcases to explore ahead :) 1. Say we have X negetive values present then if X is even then just blindly multiply them to the positive non zero values if any:
+Example : arr[] = {-2,-4,-1,-5,3,4,0} = 480 2. But say the frequencies of those X negetive values is odd then starts the real issue :)
+Example : arr[] = {-2,-1,-5,-3,-7,4,6} = Here we have to select the closet even number to the frequency of negetive values that have highest absolute value each, here it shall be -2,-3,-5,-7 as optimal choice hence -2x-3x-5x-7x4x6 = 5040
+
+#### For minimum product :
+
+**Case 1:** If there are no negetive elements in the array then it is simply printing the smallest element such that **array[index]>=0**.
+Example:
+arr[] = {2,6,4,5} = 2
+**Case 2:** If there are some negetive elements in the array then we have some subcases to explore ahead :) 1. Say we have X negetive values present then if X is odd then just blindly multiply them to the positive non zero values if any:
+Example : arr[] = {-2,-4,-1,-5,-3,4,0} = -480 2. But say the frequencies of those X negetive values is even then we have to select the closet even number to the frequency of negetive values that have highest absolute value each , Example : arr[] = {-2,-5,-3,-7,4,6} = Here , here it shall be -3,-5,-7 as optimal choice hence -3x-5x-7x4x6 = 2520
```

Comment:
This isn't formatted well. There is some syntax error. Please preview/correct it once.